src/ImageLoaders/IHex32/IntelHex32Reader.cs

```diff
@@ -0,0 +1,198 @@
+﻿#region License
+/* 
+ * Copyright (C) 2017-2018 Christian Hostelet.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+
+using System;
+using System.IO;
+using Reko.Core;
+
+namespace Reko.ImageLoaders.IntelHex32
+{
+
+    /// <summary>
+    /// A reader capable of reading a Intel Hexadecimal 32-bit format object (a.k.a. IHEX32) stream.
+    /// </summary>
+    public class IntelHex32Reader : IDisposable
+    {
+
+        #region Locals
+
+        private readonly StreamReader streamReader;
+        private int lineNum;
+        private bool IsSegmented = false;
+        private bool IsLinear = false;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructs instance of an <see cref="IntelHex32Reader" />.
+        /// </summary>
+        /// <param name="str">The source stream of the hex file.</param>
+        /// <exception cref="ArgumentNullException">If the <paramref name="str" /> is null.</exception>
+        public IntelHex32Reader(Stream str)
+        {
+            if (str == null)
+                throw new ArgumentNullException(nameof(str));
+            streamReader = new StreamReader(str);
+            lineNum = 0;
+        }
+
+        #endregion
+
+        #region IDisposable implementation
+
+        private bool disposedValue; // To detect redundant calls
```

Comment:
This field should be with the others.

Also, we're not expecting this class to be used in a multithreaded environment, but if it were, this `disposedValue` will not save you from race conditions. If racing is a concern, consider using `Interlocked.CompareExchange` on the   `streamReader` instance and nulling it out when disposing it.